# Problem Statement

There are situations when we all need to rush to hospitals for some emergency and pay high consultation chareges to the doctors. But if we have some data/important information about doctor, we can build a model to predict the fees for a doctor. 

So this is a Regression problem where we have some information related to doctors and we need to predict the consultation charges.

# Know the dataset

Features of the dataset include the following:
    
- Qualification: Qualification and degrees held by the doctor
- Experience: Experience of the doctor in number of years
- Rating: Rating given by patients
- Profile: Type of the doctor
- Miscellaneous_Info: Extra information about the doctor
- Fees: Fees charged by the doctor (Target Variable)
- Place: Area and the city where the doctor is located.

# Importing Libraries

In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import warnings
warnings.filterwarnings('ignore')
from sklearn.model_selection import train_test_split ,  cross_val_score
from sklearn.linear_model import LinearRegression
from sklearn.metrics import r2_score, mean_squared_error
from sklearn.neighbors import KNeighborsRegressor
from sklearn.ensemble import  RandomForestRegressor
from sklearn.linear_model import LinearRegression
from sklearn.ensemble import  GradientBoostingRegressor
from sklearn.linear_model import  Ridge
from sklearn.metrics import mean_squared_error, mean_absolute_error
from sklearn.preprocessing import StandardScaler
from sklearn.svm import SVR
from sklearn import metrics
from sklearn.tree import DecisionTreeRegressor


# Importing Dataset

In [6]:
ds=pd.read_excel('Final_Train.xlsx')
df_train=pd.DataFrame(ds)
print('Training Data: ')
df_train.head()

Training Data: 


Qualification           Experience Rating  \
0           BHMS, MD - Homeopathy  24 years experience   100%   
1    BAMS, MD - Ayurveda Medicine  12 years experience    98%   
2  MBBS, MS - Otorhinolaryngology   9 years experience    NaN   
3             BSc - Zoology, BAMS  12 years experience    NaN   
4                            BAMS  20 years experience   100%   

                          Place         Profile  \
0           Kakkanad, Ernakulam       Homeopath   
1         Whitefield, Bangalore        Ayurveda   
2    Mathikere - BEL, Bangalore  ENT Specialist   
3  Bannerghatta Road, Bangalore        Ayurveda   
4         Keelkattalai, Chennai        Ayurveda   

                                  Miscellaneous_Info  Fees  
0               100% 16 Feedback Kakkanad, Ernakulam   100  
1              98% 76 Feedback Whitefield, Bangalore   350  
2                                                NaN   300  
3  Bannerghatta Road, Bangalore ₹250 Available on...   250  
4              100% 4 Feedback Keelkattalai, Chennai   250

In [7]:
ds=pd.read_excel('Final_Test.xlsx')
df_test=pd.DataFrame(ds)
print('Testing Data: ')
df_test.head()

Testing Data: 


Qualification           Experience  \
0                                             MBBS  35 years experience   
1       MBBS, Diploma in Otorhinolaryngology (DLO)  31 years experience   
2                                       MBBS, DDVL  40 years experience   
3                                             BAMS   0 years experience   
4  BDS, MDS - Conservative Dentistry & Endodontics  16 years experience   

  Rating                        Place           Profile  \
0    NaN       Ghatkopar East, Mumbai  General Medicine   
1    NaN  West Marredpally, Hyderabad    ENT Specialist   
2    70%            KK Nagar, Chennai    Dermatologists   
3    NaN       New Ashok Nagar, Delhi          Ayurveda   
4   100%    Kanakpura Road, Bangalore           Dentist   

                                  Miscellaneous_Info  
0                                                NaN  
1                                                NaN  
2                   70% 4 Feedback KK Nagar, Chennai  
3                                                NaN  
4  General Dentistry Conservative Dentistry Cosme...

In [9]:
df_train.shape

(5961, 7)

We have 6 input features, 1 output variable (Fees) and 5961 observations in training data.

## Checking missing values

In [10]:
print('Missing in train data: ')
df_train.isnull().sum()

Missing in train data: 


Qualification            0
Experience               0
Rating                3302
Place                   25
Profile                  0
Miscellaneous_Info    2620
Fees                     0
dtype: int64

In [11]:
print('Missing in test data: ')
df_test.isnull().sum()

Missing in test data: 


Qualification            0
Experience               0
Rating                1090
Place                    6
Profile                  0
Miscellaneous_Info     834
dtype: int64

We can seee that some values are missing in columns Rating and Miscellaneous_Info in both training and testing data. We will be working on these missing values in further steps.